WalletWasabi/Services/WalletService.cs

```diff
@@ -486,7 +486,7 @@ private void ProcessTransaction(SmartTransaction tx, List<HdPubKey> keys = null)
 				if (foundKey != default)
 				{
 					foundKey.SetKeyState(KeyState.Used, KeyManager);
-					List<SmartCoin> spentOwnCoins = Coins.Where(x => tx.Transaction.Inputs.Any(y => y.PrevOut.N == x.Index && y.PrevOut.Hash == x.TransactionId)).ToList();
+					List<SmartCoin> spentOwnCoins = Coins.Where(x => tx.Transaction.Inputs.Any(y => y.PrevOut.Hash == x.TransactionId && y.PrevOut.N == x.Index)).ToList();
```

Comment:
This is really surprising and i cant see how it can be faster in this way. Anyway, if you checked this is faster it is okay.

Reply:
Forget this comment. I saw your test ;)

Reply:
Because indexes are low numbers, so often match, so hash equality are often executed anyway.